### PR TITLE
Documentation for trial cleaner job

### PR DESCRIPTION
**Pull request by jaroslaw-pieszka**
**Description**
Trial cleanup job documentation: overview, details, prerequisites and configuration.

Changes proposed in this pull request:

- Documentation page

**Related issue(s)**

https://github.tools.sap/kyma/backlog/issues/2730

### CODE REVIEW

File: resources/kcp/charts/kyma-environment-broker/templates/trial-cleanup-job.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ spec:
           {{ end }}
         spec:
           shareProcessNamespace: true
-          {{- with .Values.deployment.securityContext }}
+          {{- with .ValuesAPP_DRY_RUN.deployment.securityContext }}
```
Comment (wozniakjan) [inline resources/kcp/charts/kyma-environment-broker/templates/trial-cleanup-job.yaml:18]: I think this won't render as `.ValuesAPP_DRY_RUN`

**Review by wozniakjan — COMMENTED** [review]
I think there is a typo in helm chart but other than that sgtm

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+| **APP_DATABASE_SSL** | Activates the SSL mode for PostgrSQL. See all the possible values [here](https://www.postgresql.org/docs/9.1/libpq-ssl.html). | `disable`                                |
+| **APP_BROKER_URL**  | Specifies the Kyma Environment Broker URL.                                                                                    | `https://kyma-env-broker.kyma.local`     |
+| **APP_BROKER_TOKEN_URL** | Specifies the Kyma Environment Broker OAuth token endpoint.                                                                   | `https://oauth.2kyma.local/oauth2/token` |
+| **APP_BROKER_CLIENT_ID** | Specifies the username for the OAuth2 authentication in KEB.                                                                  | None                                     |
+| **APP_BROKER_CLIENT_SECRET** | Specifies the password for the OAuth2 authentication in KEB.                                                                  | None                                     |
+| **APP_BROKER_SCOPE** | Specifies the scope for the OAuth2 authentication in KEB.                                                                     | None                                     |
```
[the remark this answers is not in the document]
Comment (majakurcius) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:43]: I also tried to see whether we have a uniform way to signify no default value in such a table, but unfortunately we don't (might be something we should look into, @kyma-project/technical-writers ); various people do it differently, [using `-` or `""`](https://github.com/kyma-project/kyma/blob/33d08542b2338beb7a836200f855ac71891abc21/resources/serverless/charts/docker-registry/README.md), `N/A`,  or [not putting anything at all there](https://github.com/kyma-project/kyma/blob/8e602aea54d3c9b924f651bcc50cc62fba22ebf3/resources/ory/charts/gcloud-sqlproxy/README.md), so I don't have a specific guidelines for this for you right now.

File: resources/kcp/charts/kyma-environment-broker/templates/trial-cleanup-job.yaml (hunk 1 above)
Comment (jaroslaw-pieszka) [reply resources/kcp/charts/kyma-environment-broker/templates/trial-cleanup-job.yaml:18]: Thanks - it was accidental change.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+## Prerequisites
+
+Trial Cleanup requires access to:
+- Database to get the IDs of instances with `trial plan` which are not expired yet. 
+- Kyma Environment Broker to initiate SKR instance suspension.
+
+## Configuration
+
+The job is a cronjob with schedule that can be configured in `management-plan-config`.
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:22]: Good point.

File: resources/kcp/charts/kyma-environment-broker/templates/trial-cleanup-job.yaml (hunk 1 above)
Comment (jaroslaw-pieszka) [reply resources/kcp/charts/kyma-environment-broker/templates/trial-cleanup-job.yaml:18]: Reverted in the next commit

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# Trial Cleanup Job
+
+Trial Cleanup Job is an application that makes SKR instances with trial plan expire if those are 14 days old.
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:3]: It was just copy of documentation from another job. But you are absolutely right.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+Expiration means that SKR instance is suspended and `expired` flag is set.
+
+## Details
+
+The job can be run either in dry run mode or in production mode:
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:8]: I haven't meant `job` as Kubernetes resource but this meaning (as Kubernetes resource is more precise).

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+- In `dry run mode` information about candidate instances (i.e. instances meeting configured criteria) is logged. No changes are made.
+- In `production mode` for each instance meeting criteria PATCH request is sent to KEB. This instance is marked as `expired` and if is in `succeeded` state suspension process is started. 
+In case the instance is already in `suspended` state this instance is only marked as `expired`. 
+
+## Prerequisites
+
+Trial Cleanup requires access to:
+- Database to get the IDs of instances with `trial plan` which are not expired yet. 
+- Kyma Environment Broker to initiate SKR instance suspension.
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:18]: Now it is obvious.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+## Prerequisites
+
+The Trial Cleanup Job requires access to:
+- Database, to get the IDs of the instances with `trial` plan which are not expired yet. 
+- KEB to initiate the SKR instance suspension.
+
+## Configuration
+
+The Job is a CronJob with schedule that can be configured as the parameter in `management-plane-config`.
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:23]:
```suggestion
The Job is a CronJob with a schedule that can be [configured](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#cron-schedule-syntax) as a parameter in `management-plane-config`.
```
1. I'm still not sure what `management-plane-config` is 🤔 
2. What do you think about the idea to link to the Kubernetes documentation on CronJob schedule syntax?

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+By default, the CronJob is set to run every day at 1:15 am:
+```yaml  
+kyma-environment-broker.trialCleanup.schedule: "15 1 * * *"
+```
+
+Use the following environment variables to configure the Job:
+
+| Environment variable | Description                                                                                                                    | Default value                            |
+|---|--------------------------------------------------------------------------------------------------------------------------------|------------------------------------------|
+| **APP_DRY_RUN** | Specifies whether to run in the [`dry-run` mode](#details). dry run only logs candidate instances                              | `true`                                   |
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:33]:
```suggestion
| **APP_DRY_RUN** | Specifies whether to run in the [`dry-run` mode](#details).                                                                    | `true`                                   |
```

You explain the mode in the section you link to. Let's not duplicate content :)

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+| **APP_EXPIRATION_PERIOD** | Specifies the the [expiration period](#trial-cleanup-job) for the instances with the `trial` plan                              | `336h`                                    |
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:34]:
```suggestion
| **APP_EXPIRATION_PERIOD** | Specifies the [expiration period](#trial-cleanup-job) for the instances with the `trial` plan.                              | `336h`                                    |
```

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+If the instance is already in the `suspended` state, this instance is only marked as `expired`. 
+
+### Dry-run mode
+If you need to test your the Job, you can run it in the dry-run mode.
+In that mode, the Job only logs the information about the candidate instances (i.e. instances meeting the configured criteria). No changes are made.
+
+## Prerequisites
+
+The Trial Cleanup Job requires access to:
+- Database, to get the IDs of the instances with `trial` plan which are not expired yet.
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:18]: Can/should we specify what database? (I.e. Runtime Provisioner database)

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+Expiration means that the SKR instance is suspended and the `expired` flag is set.
+
+## Details
+
+For each instance meeting criteria, a PATCH request is sent to Kyma Environment Broker (KEB). This instance is marked as `expired`, and if it is in the `succeeded` state, the suspension process is started. 
+If the instance is already in the `suspended` state, this instance is only marked as `expired`. 
+
+### Dry-run mode
+If you need to test your the Job, you can run it in the dry-run mode.
+In that mode, the Job only logs the information about the candidate instances (i.e. instances meeting the configured criteria). No changes are made.
```
[the remark this answers is not in the document]
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:13]: I will rephrase.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md (hunk 10 above)
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:18]: Broker Database. In the next commit.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md (hunk 8 above)
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md]: Sure.

Comment (majakurcius) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md]: It's still there :D

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md (hunk 7 above)
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md]: Changed accordingly.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md (hunk 10 above)
Comment (majakurcius) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:18]: Let's make it `KEB database` then. Or `KEB Database`, if you capitalize it in your docs.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md (hunk 8 above)
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md]: It should not.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md (hunk 10 above)
Comment (jaroslaw-pieszka) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md]: Done.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+If the instance is already in the `suspended` state, this instance is just marked as `expired`. 
+
+### Dry-run mode
+If you need to test the Job, you can run it in the `dry-run` mode.
+In that mode, the Job only logs the information about the candidate instances (i.e. instances meeting the configured criteria). The instances are not affected.
+
+## Prerequisites
+
+The Trial Cleanup Job requires access to:
+- KEB database, to get the IDs of the instances with `trial` plan which are not expired yet.
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:18]:
```suggestion
- KEB database, to get the IDs of the instances with the `trial` plan which are not expired yet. 
```
This one as well.

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+
+Trial Cleanup Job is a Job that makes the SKR instances with the `trial` plan expire 14 days after their creation.
+Expiration means that the SKR instance is suspended and the `expired` flag is set.
+
+## Details
+
+For each instance meeting the criteria, a PATCH request is sent to Kyma Environment Broker (KEB). This instance is marked as `expired`, and if it is in the `succeeded` state, the suspension process is started. 
+If the instance is already in the `suspended` state, this instance is just marked as `expired`. 
+
+### Dry-run mode
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:11]:
```suggestion

### Dry-run mode
```
This one somehow managed to hide away from you. 🤔

Comment (majakurcius) [reply docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:12]: Oh no, I misplaced this new line 🙈 Sorry!

File: docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+Expiration means that the SKR instance is suspended and the `expired` flag is set.
+
+## Details
+
+For each instance meeting the criteria, a PATCH request is sent to Kyma Environment Broker (KEB). This instance is marked as `expired`, and if it is in the `succeeded` state, the suspension process is started. 
+If the instance is already in the `suspended` state, this instance is just marked as `expired`. 
+
+
+### Dry-run mode
+If you need to test the Job, you can run it in the `dry-run` mode.
```
Comment (majakurcius) [inline docs/kyma-environment-broker/03-15-trial-cleanup-cronjob.md:13]:
```suggestion
### Dry-run mode

If you need to test the Job, you can run it in the `dry-run` mode.
```
That's where the new line was supposed to go in the first place! 
Please double check :D

**Review by majakurcius — APPROVED** [review]
🚀

**Pull request merged**